### DISCUSSION TITLE
In the CMakeLists.txt file, can we extract package version from the manifest file?

### DISCUSSION THREAD

**Discussion by jsanchez19** [microsoft/vcpkg#20176] [Q&A] [answered]
If I have a library with a manifest file, where I declare its dependencies, its name, homepage, description, _and_ version. Is it possible (or in the roadmap) to be able to use this data within `CMakeLists.txt`?

Something like this..?

```CMake
vcpkg_get_version_from_manifest(VERSION_STRING)

project(my_awesome_project
  VERSION ${VERSION_STRING})
```

**Comment by mathisloge** [comment, marked as answer]
So if you have a project and using cmake >= 3.19 cmake has support to read json files. 
https://cmake.org/cmake/help/latest/command/string.html#json

**Reply by mathisloge** [reply]
e.g.
```cmake
file(READ vcpkg.json VCPKG_MANIFEST_STR)
string(JSON VCPKG_PROJ_VERSION GET ${VCPKG_MANIFEST_STR} "version")
message(STATUS "${VCPKG_PROJ_VERSION}")
project(myproj VERSION ${VCPKG_PROJ_VERSION})
```

**Reply by Be-ing** [reply]
It is good that CMake can do that but it would be nice if vcpkg did this automatically without requiring this boilerplate.

**Reply by mathisloge** [reply]
I don't know if that should be something. I mean it is like two lines instead of one.

**Reply by dg0yt** [reply]
> I mean it is like two lines instead of one.

It is probably not good to make portfiles depend on a particular manifest JSON structure and field name. And this field name already varies by port. 
ATM there are many spot where either actual versions, or reading versions from sources, are handled in portfiles and patches. This is redundant code, given that vcpkg-tool knows the version before even starting the build.

**Reply by mathisloge** [reply]
I think the use case in this question wasn't the port itself. If I have understand it correctly, the questioner has a separate project with a manifest and don't want to increase the version number both in his manifest and cmake file.

**Reply by dg0yt** [reply]
I see. In that case, your answer is the only solution: It works even when building the lib without vcpkg.

**Reply by mathisloge** [reply]
And this is definitely the important bit: 
> It works even when building the lib without vcpkg
